### ISSUE TITLE
LoRA parameters in the model have `requires_grad=False`

### ISSUE THREAD

**Issue by bhavnicksm** [huggingface/peft#245]
I loaded a opt-125M model with LoRA and tried training it on summarization with PEFT LoRA. 

It only trains about 0.5% of the model yet is slower to train than training the whole model. 

Checking the `requires_grad` for all the parameters, it shows that the LoRA params have it set to False, which is awkward. 

code: https://colab.research.google.com/drive/18XU1CaPZxt_NsRaQATe5bq_nkKIHbmoS?usp=sharing

**Comment by pacman100** [comment]
You are checking the requires_grad the wrong way, below is the correct way:
<img width="814" alt="Screenshot 2023-03-31 at 1 22 41 PM" src="https://user-images.githubusercontent.com/13534540/229059569-1dcfbe5a-8cac-4264-baf7-1fd23b55c081.png">

Also, your are checking training times on very small model wherein gains from having small number of trainable params might be getting overshadowed by the INT8 quantization overheads. Please try big model along with a minimal reproducible benchmark script to quantify training times

**Comment by bhavnicksm** [comment]
Thanks @pacman100!

I am trying out the same script with 6.7B OPT right now. Hopefully, this would show a significant training time difference.

**Comment by pacman100** [comment]
Hello, the main advantage of PEFT methods though are as follows:
1. Huge memory savings as the optimizer no longer needs to maintain states for all the params. This is how you are able to fit 6.7B param model in Colab. Full fine-tuning would certainly give OOM on Colab 16GB GPU if you try it out. Therefore, PEFT enables training LLMs on low resource hardware.
2. Tiny checkpoints which saves huge storage costs when you want to fine-tune LLMs on a lot of downstream tasks/datasets. For example, a checkpoint using PEFT method would be in MBs rather than 13GB for opt-6.7B
3. As there are less trainable params, gradient computation and communication overheads(in case of distributed training) are reduced which should make training faster. However, this is a by-product and not the main offerings of PEFT. Speed is also dependent on the hardware itself, for example, one cannot compare PEFT tuning on T4 GPU with that of full fine-tuning on A100s as A100s are multiple magnitude faster in performance.
